commit 6eb213390729706e1840eaa67124caf26ac839f3
Author: Ryan Scott <ryan.gl.scott@gmail.com>
Date:   Tue Apr 3 14:45:48 2018 -0400

    Allow building with template-haskell-2.14
    
    This requires adding some CPP to temporarily support two versions of
    GHC (8.4 and HEAD). I'll file an issue as a reminder to remove this
    CPP once we drop support for GHC 8.4.

diff --git a/src/Data/Singletons/Single/Monad.hs b/src/Data/Singletons/Single/Monad.hs
index 7da2a7b..b1928b2 100644
--- a/src/Data/Singletons/Single/Monad.hs
+++ b/src/Data/Singletons/Single/Monad.hs
@@ -8,7 +8,7 @@ This file defines the SgM monad and its operations, for use during singling.
 The SgM monad allows reading from a SgEnv environment and is wrapped around a Q.
 -}
 
-{-# LANGUAGE GeneralizedNewtypeDeriving, ParallelListComp, TemplateHaskell #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving, ParallelListComp, TemplateHaskell, CPP #-}
 
 module Data.Singletons.Single.Monad (
   SgM, bindLets, lookupVarE, lookupConE,
@@ -72,7 +72,12 @@ instance Quasi SgM where
   qReifyConStrictness = liftSgM `comp1` qReifyConStrictness
   qIsExtEnabled       = liftSgM `comp1` qIsExtEnabled
   qExtsEnabled        = liftSgM qExtsEnabled
+#if MIN_VERSION_template_haskell(2,14,0)
+  qAddForeignFilePath = liftSgM `comp2` qAddForeignFilePath
+  qAddTempFile        = liftSgM `comp1` qAddTempFile
+#else
   qAddForeignFile     = liftSgM `comp2` qAddForeignFile
+#endif
   qAddCorePlugin      = liftSgM `comp1` qAddCorePlugin
 
   qRecover (SgM handler) (SgM body) = do
diff --git a/src/Data/Singletons/Util.hs b/src/Data/Singletons/Util.hs
index 271ade3..fb64fad 100644
--- a/src/Data/Singletons/Util.hs
+++ b/src/Data/Singletons/Util.hs
@@ -11,7 +11,7 @@ Users of the package should not need to consult this file.
              TemplateHaskell, GeneralizedNewtypeDeriving,
              MultiParamTypeClasses, StandaloneDeriving,
              UndecidableInstances, MagicHash, UnboxedTuples,
-             LambdaCase, NoMonomorphismRestriction #-}
+             LambdaCase, NoMonomorphismRestriction, CPP #-}
 
 module Data.Singletons.Util where
 
@@ -417,7 +417,12 @@ instance (Quasi q, Monoid m) => Quasi (QWithAux m q) where
   qReifyConStrictness = lift `comp1` qReifyConStrictness
   qIsExtEnabled       = lift `comp1` qIsExtEnabled
   qExtsEnabled        = lift qExtsEnabled
+#if MIN_VERSION_template_haskell(2,14,0)
+  qAddForeignFilePath = lift `comp2` qAddForeignFilePath
+  qAddTempFile        = lift `comp1` qAddTempFile
+#else
   qAddForeignFile     = lift `comp2` qAddForeignFile
+#endif
   qAddCorePlugin      = lift `comp1` qAddCorePlugin
 
   qRecover exp handler = do

commit ddd353afa3c92a53c00f22919611f734eaa0f29f
Author: Ryan Scott <ryan.gl.scott@gmail.com>
Date:   Sat Jun 30 18:51:48 2018 -0400

    PNum(type (*)) requires NoStarIsType
    
    This requires adding some CPP to temporarily support two versions of
    GHC (8.4 and HEAD). I'll amend #324 as a reminder to remove this
    CPP once we drop support for GHC 8.4.

diff --git a/src/Data/Singletons/Prelude/Num.hs b/src/Data/Singletons/Prelude/Num.hs
index 9381afc..f1d00ab 100644
--- a/src/Data/Singletons/Prelude/Num.hs
+++ b/src/Data/Singletons/Prelude/Num.hs
@@ -2,6 +2,10 @@
              TypeOperators, GADTs, ScopedTypeVariables, UndecidableInstances,
              DefaultSignatures, FlexibleContexts, InstanceSigs
   #-}
+{-# LANGUAGE CPP #-}
+#if __GLASGOW_HASKELL__ >= 805
+{-# LANGUAGE NoStarIsType #-}
+#endif
 
 -----------------------------------------------------------------------------
 -- |
